docs/content/advanced/backend/entities.md

```diff
@@ -0,0 +1,158 @@
+
+@Entity()
+export class Post extends BaseEntity {
+  @Column({type: 'varchar'})
+  title: string | null;
+
+  @BeforeInsert()
+  private beforeInsert(): void {
+    this.id = generateEntityId(this.id, "post")
+  }
+}
+```
+
+This entity has one column `title` defined. However, since it extends `BaseEntity` it will also have the `id`, `created_at`, and `updated_at` columns.
+
+Medusa’s core entities all have the following format for IDs: `<PREFIX>_<RANDOM>`. For example, an order might have the ID `order_01G35WVGY4D1JCA4TPGVXPGCQM`.
+
+To generate an ID for your entity that matches the IDs generated for Medusa’s core entities, you should add a `BeforeInsert` event handler. Then, inside that handler use Medusa’s utility function `generateEntityId` to generate the ID. It accepts the ID as a first parameter and the prefix as a second parameter. The `Post` entity IDs will be of the format `post_<RANDOM>`.
+
+If you want the entity to also be soft deletable then it should extend `SoftDeletableEntity` instead:
+
+```tsx
+import { SoftDeletableEntity } from "@medusajs/medusa/dist/interfaces/models/soft-deletable-entity";
+
+@Entity()
+export class Post extends SoftDeletableEntity {
+```
+
+You can learn more about what decorators and column types you can use in [Typeorm’s documentation](https://typeorm.io/entities).
+
+### Create the Migration
+
+Additionally, you must create a migration for your entity. Migrations are used to update the database schema with new tables or changes to existing tables.
+
+You can learn more about Migrations, how to create them, and how to run them in the [Migration documentation](migrations.md).
+
+### Create a Repository
+
+Entities data can be easily accessed and modified using Typeorm [Repositories](https://typeorm.io/working-with-repository). To create a repository, create a file in `src/repositories`. For example, here’s a repository `PostRepository` that resides in `src/repositories/post.ts`:
+
+```tsx
+import { EntityRepository, Repository } from "typeorm"
+
+import { Post } from "../models/post"
+
+@EntityRepository(Post)
+export class PostRepository extends Repository<Post> { }
+```
+
+This repository is created for the `Post` and that is indicated using the decorator `@EntityRepository`.
+
+:::tip
+
+Be careful with your file names as it can cause unclear errors in Typeorm. Make sure all your file names are small letters for both entities and repositories to avoid any issues with file names.
+
+:::
+
+## Access Your Custom Entity
+
+:::note
+
+Before trying this step make sure that you’ve created and run your migrations. You also need to re-build your code using:
+
+```bash npm2yarn
+npm run build
+```
+
+:::
+
+You can access your custom entity data in the database in services or subscribers using the repository. For example, here’s a service that lists all posts:
+
+```tsx
+import { BaseService } from "medusa-interfaces";
+
+class PostService extends BaseService {
```

Comment:
Since the release will be soon, should we extend the TransactionBaseService?

Reply:
Is this in the PR #1315 or in another PR? I didn't see any reference of it

Reply:
It comes from these one https://github.com/medusajs/medusa/pull/1344